packages/multicast_dns/example/main.dart

```diff
@@ -23,7 +23,11 @@ void main() async {
     // other mDNS queries are running elsewhere on the machine.
     await for (SrvResourceRecord srv
         in client.lookup(RRType.srv, ptr.domainName)) {
-      print('Dart obvservatory instance found at ${srv.target}:${srv.port}.');
+      // Domain name will be something like "io.flutter.example@some-iphone.local._dartobservatory._tcp.local"
+      final String bundleId =
+          ptr.domainName.substring(0, ptr.domainName.indexOf('@'));
+      print(
+          'Dart obvservatory instance found at ${srv.target}:${srv.port} for "$bundleId".');
```

Comment:
more crazy formatting...

Reply:
What would you have it do? The formatter doesn't split string literals, so it has to make the best it can with what you give it. The right answer in most cases is to split the string literal into two adjacent strings and it will then format this better.

Reply:
In both cases I would override the preferred line width and not wrap.

Specifically, I would have two line widths. One would be the hard limit. If something doesn't exceed the hard limit, it wouldn't wrap it. If it does, then it would wrap it to the soft limit.

Also, if these exceeded the wrap limit, in the line 27 case I would wrap inside the expression, not after the `=` (i.e. do what it would do with a trailing comma), and in the line 29 case I would either split the string (why should the formatter _not_ do that? IMHO it should totally do that, especially if the string has a literal `\n` in it! that's a perfect place to wrap. Not that it helps here), or I would similarly act as if there was a trailing comma.

Reply:
Well FWIW, I've now split the string manually into two so this looks a little better.

Reply:
> in the line 27 case I would wrap inside the expression, not after the = 

That's a reasonable preference. I personally go back and forth on whether splitting around `=` is useful. Some LHS sides get really long (think calling very long setters), so it turns out to be useful and more users seem to be happy with the results.

> why should the formatter not do that? IMHO it should totally do that, especially if the string has a literal \n in it! that's a perfect place to wrap.

One of the goals of the formatter is to be "reversible" as much as possible. So if you:

1. Make a change to some code.

2. Format it.

3. Undo the change.

4. Format it again.

You get back to the original code. The formatter doesn't *unsplit* adjacent strings, and I think that's a good thing — users often use adjacent strings because they specifically want the split to be right where they put it. So if the formatter *did* split strings then in step 2 above, it could split a string, but then in step 4, it wouldn't put it back together again.